yaccparser.py

```diff
@@ -700,6 +701,7 @@ def p_multiplyinstruction(p):
         p[0] |= p[6]                # Set Rm
         p[0] |= p[4] << 16          # Set Rd
     else:
+        assert len(p) == 9          # Check if we have 3 registers
```

Comment:
I'm not against putting assertions for debugging purposes, but in production code, if this is something that _could_ happen (e.g. it can be a user error), you should raise a YaccError with a meaningful error message so that the user can understand what's wrong.

Reply:
Actually, you can put an explanation of the error after the condition in Python 3, like so:
`assert len(p) == 9, "Exactly 9 registers are required, but {} were found".format(len(p))`.
Maybe a better explanation than this one should be written, though.

Reply:
Yes but this will not be send to the interface

Reply:
Since all asserts are validations in the code made by us (the developers), I was thinking we could catch them in the web handler event loop and percolate them to the user interface, no?
Or is it what YaccError already do?

Reply:
Yes the purpose of defining YaccError and LexError was to be more specific about the error type, and to keep assertions only for debugging (that is, when we just want to test the code, or want to enforce that something should never happen, whatever the student is doing). This could be rethinked though